File: dashboard/src/main/java/de/dstoll/dashboard/service/EventDataReader.java
package de.dstoll.dashboard.service;

import de.dstoll.dashboard.configuration.properties.EventDataProperties;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.SpringApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Service;
import org.springframework.util.ResourceUtils;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;

@Service
class EventDataReader {

    private final ApplicationContext applicationContext;
    private final BufferedReader reader;

    EventDataReader(EventDataProperties eventDataProperties, ApplicationContext applicationContext) {
        this.applicationContext = applicationContext;
        try {
            reader = Files.newBufferedReader(ResourceUtils.getFile(eventDataProperties.getLocation()).toPath());
        } catch (FileNotFoundException fnfex) {
            throw new RuntimeException("Event data file not found!");
        } catch (IOException e) {
            throw new RuntimeException("Fehler beim Einlesen der Event-Daten!");
        }
    }

    public String readData() {
        try {
            String line = reader.readLine();
            if(line != null) {
                return line;
            } else {
                reader.close();
                System.exit(SpringApplication.exit(applicationContext));
            }
        } catch (IOException iox) {
            throw new RuntimeException("Fehler beim Einlesen der Event-Daten!");
        }
        return null;
    }

}

File: kafka-producer/src/main/kotlin/de/dstoll/kafkaproducer/service/EventDataReader.kt
package de.dstoll.kafkaproducer.service

import de.dstoll.kafkaproducer.configuration.properties.EventDataProperties
import org.springframework.boot.SpringApplication
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Service
import org.springframework.util.ResourceUtils
import java.io.BufferedReader
import java.io.FileReader
import java.io.IOException
import java.lang.RuntimeException
import java.util.zip.ZipFile
import kotlin.system.exitProcess

@Service
class EventDataReader(
        private val eventDataProperties: EventDataProperties,
        private val applicationContext: ApplicationContext
) {

    private val reader: BufferedReader = ResourceUtils.getFile(eventDataProperties.location).bufferedReader()

    fun readData(): String {
        return try {
            val line = reader.readLine()
            if(line != null) {
                line
            } else {
                reader.close()
                exitProcess(SpringApplication.exit(applicationContext))
            }
        } catch (iox: IOException) {
            reader.close()
            throw RuntimeException("Fehler beim Einlesen der Event-Daten!")
        }
    }

}